test/support_archive/support_archive_test.go

```diff
@@ -0,0 +1,120 @@
+	"github.com/Dynatrace/dynatrace-operator/test/oneagent"
+	"github.com/Dynatrace/dynatrace-operator/test/operator"
+	"github.com/Dynatrace/dynatrace-operator/test/setup"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
+	"sigs.k8s.io/e2e-framework/klient/conf"
+	"sigs.k8s.io/e2e-framework/pkg/env"
+	"sigs.k8s.io/e2e-framework/pkg/envconf"
+	"sigs.k8s.io/e2e-framework/pkg/features"
+)
+
+var testEnvironment env.Environment
+
+func TestMain(m *testing.M) {
+	workingDir, err := os.Getwd()
+	if err != nil {
+		fmt.Println(err)
+		return
+	}
+	//	require.NoError(m, err)
+
+	fmt.Println(workingDir)
+
+	testEnvironment = environment.Get()
+	testEnvironment.BeforeEachTest(dynakube.DeleteIfExists(dynakube.NewBuilder().WithDefaultObjectMeta().Build()))
+	testEnvironment.BeforeEachTest(oneagent.WaitForDaemonSetPodsDeletion())
+	testEnvironment.BeforeEachTest(namespace.Recreate(namespace.NewBuilder(dynakube.Namespace).Build()))
+
+	testEnvironment.AfterEachTest(dynakube.DeleteIfExists(dynakube.NewBuilder().WithDefaultObjectMeta().Build()))
+	testEnvironment.AfterEachTest(oneagent.WaitForDaemonSetPodsDeletion())
+	testEnvironment.AfterEachTest(namespace.Delete(dynakube.Namespace))
+
+	testEnvironment.Run(m)
+}
+
+func TestSupportArchive(t *testing.T) {
+	testEnvironment.Test(t, SupportArchiveExecution(t))
+}
+
+func SupportArchiveExecution(t *testing.T) features.Feature {
+	secretConfig := getSecretConfig(t)
+
+	supportArchiveExecution := features.New("support archive execution")
+	setup.InstallOperator(supportArchiveExecution, secretConfig)
+	setup.AssessOperatorDeployment(supportArchiveExecution)
+	supportArchiveExecution.Assess("support archive subcommand can be executed correctly", checkSupportArchiveExecution())
+	return supportArchiveExecution.Feature()
+}
+
+func checkSupportArchiveExecution() features.Func {
+	return func(ctx context.Context, t *testing.T, environmentConfig *envconf.Config) context.Context {
+		result := executeSupportArchive(context.TODO(), t, environmentConfig, "--stdout")
+		require.NotNil(t, result)
+
+		zipReader, err := gzip.NewReader(result.StdOut)
+		require.NoError(t, err)
+
+		tarReader := tar.NewReader(zipReader)
+		hdr, err := tarReader.Next()
+		require.NoError(t, err)
+		assert.Equal(t, "operator-version.txt", hdr.Name)
+
+		resultString := make([]byte, hdr.Size)
+		resultLen, err := tarReader.Read(resultString)
+		require.Equal(t, io.EOF, err)
+		assert.Equal(t, hdr.Size, int64(resultLen))
+
+		return ctx
+	}
+}
+
+func executeSupportArchive(ctx context.Context, t *testing.T, environmentConfig *envconf.Config, cmdLineArguments string) *pod.ExecutionResult {
+	resources := environmentConfig.Client().Resources()
+	pods := operator.Get(t, ctx, resources)
+	for _, podItem := range pods.Items {
+		require.NotNil(t, podItem)
+
+		if strings.Contains(podItem.Name, "dynatrace-operator") {
```

Comment:
Nit: `functional.Filter(pod.Items, dynatraceOperatorMatch)` could be used before the loop.
As a result, loop's body is not if-nested.